crates/sui-types/src/digests.rs

```diff
@@ -338,11 +338,19 @@ impl fmt::UpperHex for CheckpointDigest {
 }
 
 impl std::str::FromStr for CheckpointDigest {
-    type Err = anyhow::Error;
+    type Err = crate::error::SuiError;
```

Comment:
Can we instead use a more specific error for this instead of the kitchen sync that is SuiError?